### PR TITLE
update docs for 2.6

### PR DESCRIPTION
**Pull request by sjsrey**

### CODE REVIEW

File: docs/_static/documentation_options.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
 var DOCUMENTATION_OPTIONS = {
     URL_ROOT: document.getElementById("documentation_options").getAttribute('data-url_root'),
-    VERSION: '2.4.0',
+    VERSION: '2.2.0+217.geefab596.dirty',
```
Comment (jGaboardi) [inline docs/_static/documentation_options.js:3]: Still seeing the `dirty` version, but this we be cleared up with the next, push, correct? 

@martinfleis

Comment (martinfleis) [reply docs/_static/documentation_options.js:3]: @sjsrey you need to merge master into this branch and run sphinx again to get the correct version.

**Comment by knaaptime** [comment]
i think this can be trimmed down a bit. Since the docs build in the cloud, we dont need any of the generated files inside the repo, just the source rsts

File: docs/_static/documentation_options.js (hunk 1 above)
Comment (knaaptime) [reply docs/_static/documentation_options.js:3]: i think this is just a remnant from the last time the docs were build locally on serges machine, this gets overwritten in the gh-pages branch when the docs are built as part of GHA

**Comment by knaaptime** [comment]
do a `make clean` and push again

**Comment by sjsrey** [comment]
after cutting the release and fetching, when I make clean; make html I'm getting (locally):

image

**Comment by knaaptime** [comment]
yeah maybe im wrong about this. We get the same ugliness on the live segregation docs at pysal.org/segregation

**Comment by sjsrey** [comment]
That was due to work on the docs happening after the release commits.

**Comment by sjsrey** [comment]
Maybe the fix is to publish out of gh-pages branch rather than master (which pysal is doing now). this way when we push to master the docs are not rebuild everytime. we update the docs only when pushing to gh-pages branch?

**Comment by sjsrey** [comment]
> Maybe the fix is to publish out of gh-pages branch rather than master (which pysal is doing now). this way when we push to master the docs are not rebuild everytime. we update the docs only when pushing to gh-pages branch?

I thing gpd is wrestling with the same:

image

**Comment by sjsrey** [comment]
For now, I'm ok with the docs pointing to the latest master. (Rather than the -2 version they are pointing to now). So I would say we merge this, then have a dev discussion about possible approaches to the docs and versioning across the ecosystem (on Friday).

**Comment by martinfleis** [comment]
The `.dirty` issue is something we haven't figured out in geopandas. We use the same RTD settings for geopandas, xyzservices and momepy and the issue shows up only in geopandas. No clue why.

**Comment by knaaptime** [comment]
is geopandas actually using RTD? we're only using gh-pages here

i think it may have to do with whether you leave the pre-built version of the docs in the repo or not. When i cooked up the gh-pages recipe about a year ago, the goal was to get rid of the docs/docsrc combo, so i was also trying to be as sparse as possible in the docs dir (i.e. i try to make sure to do a `make clean` before pushing so that the files generated by sphinx arent included. The GHA script runs after the commit, so it generates all those extra files in `_build` before publishing up to gh-pages. From git's perspective, those additional files mean that the git history is dirty because the generated files arent part of the last commit

**Comment by knaaptime** [comment]
my guess, anyway

**Comment by knaaptime** [comment]
...which makes me realize we might want to include the doc building [action](https://github.com/knaaptime/segregation/blob/master/.github/workflows/build_docs.yml) in momepy, which would ensure that the docs are published at pysal.org/momepy for consistency

**Comment by martinfleis** [comment]
> is geopandas actually using RTD? we're only using gh-pages here

Yes, we no longer use `gh-pages`, everything is RTD now, meaning that there's nothing pre-built (the build folder is in `.gitignore`). But that is the same across those repos I mentioned.

> ...which makes me realize we might want to include the doc building action in momepy, which would ensure that the docs are published at pysal.org/momepy for consistency

as a duplication or instead of `docs.momepy.org`? I like the dedicated domain :).

**Comment by knaaptime** [comment]
cool, your call obviously. I just default to consistency across the subpackages

**Comment by martinfleis** [comment]
We can try making an alias, to have both.

**Comment by knaaptime** [comment]
i think the issue is [here](https://github.com/pysal/momepy/blob/0a3d34da77faa63babcd2015e71f71c4b88d768c/.gitignore#L69)

momepy also includes `doc/generated` in gitignore but geopandas doesnt. That gets created by sphinx resulting in the dirty tag

**Comment by martinfleis** [comment]
Hmm, not sure. GeoPandas has the equivalent [here](https://github.com/geopandas/geopandas/blob/a80081e52ed3bd069a75a4caaed44e45e35c647d/.gitignore#L65). The generated docs are at a different place. If you try running `make html` locally, git does not see any change.

One difference I realised now is that `momepy` and `xyzservices` are not using conda on RTD. That may be the key difference.

**Comment by martinfleis** [comment]
Here - https://github.com/readthedocs/readthedocs.org/issues/8201

**Comment by knaaptime** [comment]
nice find

**Comment by knaaptime** [comment]
aside, is there a reason you guys prefer RTD to gh-pages? just the versioning feature? I ditched the former as soon as possible because its got these awkward fragility issues

**Comment by martinfleis** [comment]
I don't think we ever had this discussion properly. Surely not recently in the era of GHA. But it is true that it can be tricky for geospatial packages. That is a reason why momepy notebooks are not executed on `make html` but are pre-executed. I just wasn't able to fit under the RTD memory limit, so momepy might be another candidate for moving to gh-pages. With geopandas we haven't had issues recently so there was no need to think about it.

**Pull request merged**